### PR TITLE
[Doc] Add documentation for GGUF quantization

### LINKED ISSUE #7978: [Usage]: run gguf model need template，how to write？
### Your current environment

BadRequestError: Error code: 400 - {'object': 'error', 'message': 'As of transformers v4.44, default chat template is no longer allowed, so you must provide a chat template if the tokenizer does not define one.', 'type': 'BadRequestError', 'param': None, 'code': 400}

### How would you like to use vllm

 CUDA_VISIBLE_DEVICES=1 vllm serve  /ai/qwen1.5-1.8b.gguf --host 0.0.0.0  --port 10868 --max-model-len 4096   --trust-remote-code --tensor-parallel-size 1  --dtype=half --quantization gguf --load-format gguf

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### LINKED ISSUE #8401: [Usage]: how to use openai compatible api to run GGUF model?
### Your current environment

```text
The output of `python collect_env.py`
```

### How would you like to use vllm

I want to run inference of a [specific model](put link here). I don't know how to integrate it with vllm.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### LINKED ISSUE #8570: [Usage]: Use GGUF model with docker when hf repo has multiple quant versions
### Your current environment

I skipped the collect_env step as I use the latest docker container `v0.6.1.post2` of vllm.

### How would you like to use vllm

I want to use a GGUF variant of the [Mistral Large Instruct 2407](https://huggingface.co/mistralai/Mistral-Large-Instruct-2407) model with vllm inside a docker container. I followed [the docs](https://docs.vllm.ai/en/latest/serving/deploying_with_docker.html) for setting up a container.  
The repos listed under the [quantized category](https://huggingface.co/models?other=base_model:quantized:mistralai/Mistral-Large-Instruct-2407) of the model are all GGUF, each with multiple different quant versions inside them. Only 2 of the repos have a `config.json` ([this](https://huggingface.co/gaianet/Mistral-Large-Instruct-2407-GGUF/tree/main) and [this](https://huggingface.co/second-state/Mistral-Large-Instruct-2407-GGUF/tree/main)). How can I tell vllm which quantized version of a repo I want to use? 
Info: I use an A100 80GB.   

What I tried:
> docker run  --gpus all --name vllm -v /mnt/disk1/hf_models:/root/.cache/huggingface --env "HUGGING_FACE_HUB_TOKEN=<my_token>" -p 8080:8000 --ipc=host vllm/vllm-openai:latest \
--model bartowski/Mistral-Large-Instruct-2407-GGUF --tokenizer mistralai/Mistral-Large-Instruct-2407 --gpu-memory-utilization 0.98

Result: 
> ValueError: No supported config format found in bartowski/Mistral-Large-Instruct-2407-GGUF

Then I tried one of the repos that have a config.json:
> docker run  --gpus all --name vllm -v /mnt/disk1/hf_models:/root/.cache/huggingface --env  "HUGGING_FACE_HUB_TOKEN=<my_token>" -p 8080:8000 --ipc=host vllm/vllm-openai:latest \
--model second-state/Mistral-Large-Instruct-2407-GGUF --tokenizer mistralai/Mistral-Large-Instruct-2407 --gpu-memory-utilization 0.98

Result: 
> torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 288.00 MiB. GPU 0 has a total capacity of 79.15 GiB of which 213.69 MiB is free. Process 981263 has 78.93 GiB memory in use. [...]

Info: No other process ran on the GPU, the memory was empty before.

So it seems at least that vllm tries to load *something*. But how can I specifiy *which* quantized version I want to load? E.g. the q4_K_S variant? I tried giving a link (`--model https://huggingface.co/bartowski/Mistral-Large-Instruct-2407-GGUF/tree/main/Mistral-Large-Instruct-2407-Q4_K_M`), but it seems `--model` only accepts the HF `repo/model` format.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by Isotr0py**
FILL IN THE PR DESCRIPTION HERE

FIX #8570 
FIX #8401 
FIX #7978 (*link existing issues this PR will resolve*)

This PR adds GGUF quantization documents for some frequently asked basic usage issues like: 
- #8570
- #8401 
- #7978

**BEFORE SUBMITTING, PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**

---

<details>

<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to vLLM! Before submitting the pull request, please ensure the PR meets the following criteria. This helps vLLM maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Frontend]</code> For changes on the vLLM frontend (e.g., OpenAI API server, <code>LLM</code> class, etc.) </li>
    <li><code>[Kernel]</code> for changes affecting CUDA kernels or other compute kernels.</li>
    <li><code>[Core]</code> for changes in the core vLLM logic (e.g., <code>LLMEngine</code>, <code>AsyncLLMEngine</code>, <code>Scheduler</code>, etc.)</li>
    <li><code>[Hardware][Vendor]</code> for hardware-specific changes. Vendor name should appear in the prefix (e.g., <code>[Hardware][AMD]</code>).</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a> and <a href="https://google.github.io/styleguide/cppguide.html">Google C++ style guide</a>.</li>
    <li>Pass all linter checks. Please use <a href="https://github.com/vllm-project/vllm/blob/main/format.sh"><code>format.sh</code></a> to format your code.</li>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li>Include sufficient tests to ensure the project to stay correct and robust. This includes both unit tests and integration tests.</li>
    <li>Please add documentation to <code>docs/source/</code> if the PR modifies the user-facing behaviors of vLLM. It helps vLLM user understand and utilize the new features or changes.</li>
</ul>

<h3>Adding or changing kernels</h3>
<p>Each custom kernel needs a schema and one or more implementations to be registered with PyTorch.</p>
<ul>
    <li>Make sure custom ops are registered following PyTorch guidelines: <a href="https://pytorch.org/tutorials/advanced/cpp_custom_ops.html#cpp-custom-ops-tutorial">Custom C++ and CUDA Operators</a> and <a href="https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU">The Custom Operators Manual</a></li>
    <li>Custom operations that return <code>Tensors</code> require meta-functions. Meta-functions should be implemented and registered in python so that dynamic dims can be handled automatically. See above documents for a description of meta-functions.</li>
    <li>Use <a href="https://pytorch.org/docs/stable/library.html#torch.library.opcheck"><code>torch.libary.opcheck()</code></a> to test the function registration and meta-function for any registered ops.  See <code>tests/kernels</code> for examples.</li>
    <li>When changing the C++ signature of an existing op, the schema must be updated to reflect the changes.</li>
    <li>If a new custom type is needed, see the following document: <a href="https://docs.google.com/document/d/18fBMPuOJ0fY5ZQ6YyrHUppw9FA332CpNtgB6SOIgyuA">Custom Class Support in PT2</a>.
</ul>

<h3>Notes for Large Changes</h3>
<p>Please keep the changes as concise as possible. For major architectural changes (>500 LOC excluding kernel/data/config/test), we would expect a GitHub issue (RFC) discussing the technical design and justification. Otherwise, we will tag it with <code>rfc-required</code> and might not go through the PR.</p>

<h3>What to Expect for the Reviews</h3>

<p>The goal of the vLLM team is to be a <i>transparent reviewing machine</i>. We would like to make the review process transparent and efficient and make sure no contributor feel confused or frustrated. However, the vLLM team is small, so we need to prioritize some PRs over others. Here is what you can expect from the review process: </p>

<ul>
    <li> After the PR is submitted, the PR will be assigned to a reviewer. Every reviewer will pick up the PRs based on their expertise and availability.</li>
    <li> After the PR is assigned, the reviewer will provide status update every 2-3 days. If the PR is not reviewed within 7 days, please feel free to ping the reviewer or the vLLM team.</li>
    <li> After the review, the reviewer will put an <code> action-required</code> label on the PR if there are changes required. The contributor should address the comments and ping the reviewer to re-review the PR.</li>
    <li> Please respond to all comments within a reasonable time frame. If a comment isn't clear or you disagree with a suggestion, feel free to ask for clarification or discuss the suggestion.
 </li>
</ul>

<h3>Thank You</h3>

<p> Finally, thank you for taking the time to read these guidelines and for your interest in contributing to vLLM. Your contributions make vLLM a great tool for everyone! </p>

</details>

### CODE REVIEW

**Review by mgoin — APPROVED** [review]
Thank you for addressing the issues, this is a nice start for gguf. It would be nice to show the TP support and maybe known lack of support for other features.

File: docs/source/quantization/gguf.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+   Currently, vllm only supports loading single-file GGUF models. If you have a multi-files GGUF model, you can use `gguf-split <https://github.com/ggerganov/llama.cpp/pull/6135>`_ tool to merge them to a single-file model.
+
+To run a GGUF model with vLLM, you can download and use the local GGUF model from `TheBloke/TinyLlama-1.1B-Chat-v1.0-GGUF <https://huggingface.co/TheBloke/TinyLlama-1.1B-Chat-v1.0-GGUF>`_ with the following command:
+
+.. code-block:: console
+
+   $ wget https://huggingface.co/TheBloke/TinyLlama-1.1B-Chat-v1.0-GGUF/resolve/main/tinyllama-1.1b-chat-v1.0.Q4_K_M.gguf
+   $ # We recommend using the tokenizer from base model to avoid long-time and buggy tokenizer conversion.
+   $ vllm serve --model ./tinyllama-1.1b-chat-v1.0.Q4_K_M.gguf --tokenizer TinyLlama/TinyLlama-1.1B-Chat-v1.0
```
Comment (russellb) [inline docs/source/quantization/gguf.rst:20]: When I run this command, I get an error

```
$ vllm serve --model ./tinyllama-1.1b-chat-v1.0.Q4_K_M.gguf --tokenizer TinyLlama/TinyLlama-1.1B-Chat-v1.0
usage: vllm serve <model_tag> [options]
vllm serve: error: the following arguments are required: model_tag
```

Comment (russellb) [reply docs/source/quantization/gguf.rst:20]:
```suggestion
   $ vllm serve ./tinyllama-1.1b-chat-v1.0.Q4_K_M.gguf --tokenizer TinyLlama/TinyLlama-1.1B-Chat-v1.0
```

**Review by russellb — CHANGES_REQUESTED** [review]

**Review by russellb — APPROVED** [review]
lgtm - tested both the commands and sample code

**Pull request merged**